src/ResourceManager/Sql/Commands.Sql/Auditing/Services/SqlAuditAdapter.cs

```diff
@@ -470,7 +470,7 @@ private BlobAuditingCreateOrUpdateParameters PolicizeBlobAuditingModel(BaseBlobA
             if (!IgnoreStorage && (model.AuditState == AuditStateType.Enabled))
             {
                 properties.StorageEndpoint = ExtractStorageAccountName(model, storageEndpointSuffix);
-                properties.StorageAccountAccessKey = ExtractStorageAccountKey(model.StorageAccountName);
+                properties.StorageAccountAccessKey = ExtractStorageAccountKey(model.StorageAccountName, model.StorageKeyType);
```

Comment:
we need tests for this change

Reply:
Added a check for AuditAction value. Regarding the storage account access key, the value is not returned thus it cannot be checked.